### PR TITLE
Learn: Introduction to service mesh using linkerd

### PR DESCRIPTION
**Pull request by Blakelist7**
**Description**
This PR adds the course for Introduction to Service Mesh using linkerd.
Chapter1 - Getting Started 
#1986 
 
**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Blakelist7** [comment]
@Nikhil-Ladha Could you please verify the first chapter `getting-started`?
Do let me know if it needs any modification.

**Comment by Nikhil-Ladha** [comment]
Have you tried to build the site locally and confirmed if things are working?
The build is failing due to improper closing of tags in the MDX file.

File: content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx [hunk 1]
```diff
@@ -0,0 +1,114 @@
+---
+doctype: "Chapter"
+chapterTitle: "Getting Started"
+description: "Meshery is the service mesh managemen plane which offers lifecycle, configuration, and performance management of service meshes and their workloads."
+videos:
```
Comment (Nikhil-Ladha) [inline content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx:5]: Add a dummy value, don't keep it empty

Comment (Blakelist7) [reply content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx]: 👍🏽

**Comment by Blakelist7** [comment]
> Have you tried to build the site locally and confirmed if things are working?
Yes

> The build is failing due to improper closing of tags in the MDX file
Done.
Please check now

**Comment by l5io** [comment]
🚀 Preview for commit 09655cf74e34899cf22b19fb633f5d0169640033 at: https://60ed94dc8ac92c409bb6730b--layer5ng.netlify.app

**Comment by Blakelist7** [comment]
> The styling and text are not correct, check the `getting-started` file for istio and update that. For example the headings.

Made some changes with reference to istio.
Check now, if anything needs some tweaking

**Comment by l5io** [comment]
🚀 Preview for commit 710251cfeb23e189c439b4337529e4db39724a0b at: https://60edb8a0faeeb208951ccec2--layer5ng.netlify.app

File: content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+import { ChapterStyle } from "../../../../src/components/Learn-Components/Chapters-Style/chapters.style.js";
+import LinkerdAdapter from "../../../../src/assets/images/learn-layer5/linkerd/linkerd-adapter.png";
+import InstallLinkerd from "../../../../src/assets/images/learn-layer5/linkerd/install-linkerd.png";
+
+<ChapterStyle>
+
+<h2 class="chapter-sub-heading">Deploy Linkerd</h2>
+
+With <a href="https://github.com/layer5io/linkerd-service-mesh-workshop/blob/master/prereq/README.md">prerequisites</a> complete, you are ready to download and deploy Linkerd resources.
```
Comment (Nikhil-Ladha) [inline content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx:17]: There shouldn't be any reference to the Github repo.

Comment (Nikhil-Ladha) [reply content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx:17]: You can change the sentence to what we have in istio, that will work too.

File: content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+In the Linkerd management page:
+
+1. Type `linkerd` into the namespace field.
+1. Click the (+) icon on the `Install` card and select `Latest Linkerd` to install the latest version of Linkerd.
+
+   <a href={InstallLinkerd}>
+     <img src={InstallLinkerd} align="center" />
+   </a>
+
+<h3>For manual steps go <a href="https://github.com/layer5io/linkerd-service-mesh-workshop/blob/master/lab-1/README.md#appendix">here</a></h3>
```
Comment (Nikhil-Ladha) [inline content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx:44]: No reference to github

File: content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+<h4 class="chapter-alt-heading">
+  <a name="2">Verify install</a>
+</h4>
+
+Linkerd is deployed in the `linkerd` Kubernetes namespace. Verify that Linkerd and its components are deployed, execute the command:
+
+```sh
+linkerd check
+```
```
Comment (Nikhil-Ladha) [inline content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx:57]: Move this before the `Manual Installation` section above

File: content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx (hunk 3 above)
Comment (Nikhil-Ladha) [reply content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx:44]: Move this after the section below. (Explained below)
Also, don't add a link to the section rather move the `Alternative Manual Install` section here.
Then, you will have 2 `Verify Install` sections, remove the one under the `Alternative Section`.

File: content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx (hunk 4 above)
Comment (Nikhil-Ladha) [reply content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx:57]: Also, these headings should use `h3` with class `chapter-sub-heading`, only the ones under `Manual installation` will have `h4`

File: content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+<h4 class="chapter-alt-heading">
+  <a>Enforce mTLS strict mode</a>
+</h4>
```
Comment (Nikhil-Ladha) [inline content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx:61]: Use `h3`, and class `chapter-sub-heading`

File: content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+By establishing mutually-authenticated connections between Linkerd proxies , Linkerd automatically enables mutual Transport Layer Security (mTLS) by default for most HTTP-based communication between services.
+
+<h4 class="chapter-alt-heading">
+  <a name="3">Confirming Add-ons</a>
+</h4>
```
Comment (Nikhil-Ladha) [inline content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx:67]: Same as above

File: content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+  3. Jaeger <a href="https://www.jaegertracing.io/">(site)</a>
+  4. Dashboard <a href="https://linkerd.io/2.10/reference/architecture/#dashboard">(site)</a>
+
+You will use Meshery for collecting and viewing metrics, and Jaeger for viewing distribued traces.
+
+<br />
+
+Alternative, manual installation steps are provided for reference below. No need to execute these if you have performed the steps above.
+
+<h3 class="chapter-sub-heading">
```
Comment (Nikhil-Ladha) [inline content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx:82]: This heading will not have any class.

**Review by Nikhil-Ladha — CHANGES_REQUESTED** [review]

**Comment by Nikhil-Ladha** [comment]
The first one is a bit tiring, but then it will become easier.

**Comment by Nikhil-Ladha** [comment]
Also, please don't push the changes for all the chapters in 1 PR. Send out a PR for each chapter, otherwise, it's very very difficult to review the changes.

File: content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx (hunk 2 above)
Comment (Blakelist7) [reply content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx:17]: ok

**Comment by l5io** [comment]
🚀 Preview for commit fecc1ce217cb511f8021c1b09eff806f6b7f58d7 at: https://60edece562233550d49f63d2--layer5ng.netlify.app

**Comment by Blakelist7** [comment]
Done :)

File: content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+lectures: 8
+---
+
+import { ChapterStyle } from "../../../../src/components/Learn-Components/Chapters-Style/chapters.style.js";
+import LinkerdAdapter from "../../../../src/assets/images/learn-layer5/linkerd/linkerd-adapter.png";
+import InstallLinkerd from "../../../../src/assets/images/learn-layer5/linkerd/install-linkerd.png";
+
+<ChapterStyle>
+
+<h2 class="chapter-sub-heading">Deploy Linkerd</h2>
```
Comment (Nikhil-Ladha) [inline content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx:15]: Let's make it `Setup Linkerd` to be consistent.

File: content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+In the Linkerd management page:
+
+1. Type `linkerd` into the namespace field.
+1. Click the (+) icon on the `Install` card and select `Latest Linkerd` to install the latest version of Linkerd.
+
+   <a href={InstallLinkerd}>
+     <img src={InstallLinkerd} align="center" />
+   </a>
+
+<h3 class="chapter-sub-heading">
```
Comment (Nikhil-Ladha) [inline content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx:44]: Move this section after the manual installation section below.

File: content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+</h3>
+
+Linkerd is deployed in the `linkerd` Kubernetes namespace. Verify that Linkerd and its components are deployed, execute the command:
+
+```sh
+linkerd check
+```   
+
+<h3>
+  <a>Appendix - Alternative Manual Install</a>
```
Comment (Nikhil-Ladha) [inline content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx:55]: `a` tag not needed.

**Review by Nikhil-Ladha — COMMENTED** [review]
Once these are done, we are good to go 👌🏼

File: content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx (hunk 9 above)
Comment (Blakelist7) [reply content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx:44]: `Verify install` section below `Manual Install` section, right?

Comment (Nikhil-Ladha) [reply content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx:44]: Yup.

**Comment by Blakelist7** [comment]
> Once these are done, we are good to go 👌🏼

Done.

File: content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+In the Linkerd management page:
+
+1. Type `linkerd` into the namespace field.
+1. Click the (+) icon on the `Install` card and select `Latest Linkerd` to install the latest version of Linkerd.
+
+   <a href={InstallLinkerd}>
+     <img src={InstallLinkerd} align="center" />
+   </a>   
+
+<h3>Appendix - Alternative Manual Install</h3>
```
Comment (Nikhil-Ladha) [inline content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx:44]: Last one promise :p
Remove the `Appendix - `

Comment (Nikhil-Ladha) [reply content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx:44]: And make it `Alternative:Manual Installation`

Comment (Blakelist7) [reply content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx]: :')

Comment (Blakelist7) [reply content-learn/mastering-meshery-istio/introduction-to-service-meshes/linkerd/getting-started.mdx]: Done

**Comment by l5io** [comment]
🚀 Preview for commit 9ee3d4d0af5b50da4c9711c5e7b92cc061ee6c7c at: https://60ee9ee3c125e8215ff82424--layer5ng.netlify.app

**Comment by Blakelist7** [comment]
Done

**Review by Nikhil-Ladha — APPROVED** [review]
Nice 👍🏼 . Hopefully, in next PR it will become easier for you :)

**Pull request merged**